src/Arcus.WebApi.Logging/ExceptionHandlingMiddleware.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Arcus.WebApi.Logging
+{
+    public class ExceptionHandlingMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly Func<string> _getLoggingCategory;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ExceptionHandlingMiddleware"/> class.
+        /// </summary>
+        /// <param name="next"></param>
+        public ExceptionHandlingMiddleware(RequestDelegate next)
+            : this(next, string.Empty)
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ExceptionHandlingMiddleware"/> class.
+        /// </summary>
+        /// <param name="next"></param>
+        /// <param name="categoryName">The category-name for messages produced by the logger.</param>
+        public ExceptionHandlingMiddleware(RequestDelegate next, string categoryName)
+            : this(next, () => categoryName)
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ExceptionHandlingMiddleware"/> class.
+        /// </summary>
+        /// <param name="next"></param>
+        /// <param name="getLoggingCategory">The function that returns the category-name that must be used by the logger
+        /// when writing log messages.</param>
+        public ExceptionHandlingMiddleware(RequestDelegate next, Func<string> getLoggingCategory)
+        {
```

Comment:
Maybe good to guard for getLoggingCategory being null?

Reply:
Well, I'd like to avoid throwing an exception for that so I've solved it during logging itself.  If categoryName resolves to null, it gets replaced with an empty string.

Reply:
That's also ok for us